TypeError: consumeBody(...).clone is not a function

I have this markdown content:

```md
![barley](//images.ctfassets.net/kpzg83k38b4x/7EBkxPB3EDz0wJKRJhEq3q/916f98bd010edc03c87e82e5c1e79a7b/barley.jpg)
```

... and I init `markdown-it` with the plugin like so:

```js
import MarkdownIt from 'markdown-it'
import { markdownItImageSize } from 'markdown-it-image-size'

const md = new MarkdownIt('default', {
  linkify: true,
  breaks: true,
})

md.use(markdownItImageSize)
```

I'm getting this `TypeError`:

```
TypeError: consumeBody(...).clone is not a function
    at SyncResponse.buffer (browser.js:287:30)
    at getImageDimensionsFromExternalImage (index.js:1428:27)
    at md.renderer.rules.image (index.js:1392:15)
    at Renderer.renderInline (renderer.js:275:23)
    at Renderer.render (renderer.js:330:22)
    at MarkdownIt.render (index.js:544:24)
```

Any ideas what might be causing it?

Hi, thanks for the quick response!

Yeah, it doesn't have a protocol as that's what I'm getting from the API (Contentful in this case). But it made no difference when the protocol was added, forgot to mention.

I've updated to `markdown-it-image-size@next` and am using `markdown-it@13.0.1`, but I'm still getting the error - looks like it's coming from `sync-fetch`:

<img alt="image" src="https://user-images.githubusercontent.com/1656595/200171939-de914532-ff50-456d-ba76-903ccb261200.png">

... which is this method in its `browser.js`:

<img alt="image" src="https://user-images.githubusercontent.com/1656595/200172114-1769ddcc-9b1f-4eda-a849-44eae5360643.png">

By the way, I'm using this in a Nuxt 2 static site, and when I tried building it with the `next` version I also got this error:

```
3:02:59 PM: $ npm run generate
3:02:59 PM: > generate
3:02:59 PM: > nuxt generate
3:03:11 PM: [fatal] Nuxt build error
3:03:11 PM:   ERROR in ./node_modules/markdown-it-image-size/dist/index.js
3:03:11 PM:   Module not found: Error: Can't resolve 'fs' in 'node_modules/markdown-it-image-size/dist'
  @ ./node_modules/markdown-it-image-size/dist/index.js 5:17-30
3:03:11 PM:   @ ./plugins/markdown-it.js
3:03:11 PM:   @ ./node_modules/.cache/nuxt/index.js
3:03:11 PM:   @ ./node_modules/.cache/nuxt/client.js
3:03:11 PM:   @ multi ./node_modules/@nuxt/components/lib/installComponents.js ./node_modules/.cache/nuxt/client.js
```

For context, this is what I'm doing in `plugins/markdown-it.js`:

```js
import MarkdownIt from 'markdown-it'
import { markdownItImageSize } from 'markdown-it-image-size'

const addLazyAttribute = md => {
  md.renderer.rules.image = function (tokens, idx, options, env, self) {
    const token = tokens[idx]
    const src = token.attrGet('src')
    const alt = md.utils.escapeHtml(token.content)

    return `<img src="${src}" alt="${alt}" loading="lazy">`
  }
}

export default ({}, inject) => {
  const md = new MarkdownIt('default', {
    linkify: true,
    breaks: true,
  })

  md
    .use(addLazyAttribute)
    .use(markdownItImageSize)

  // Make $md available in Nuxt pages
  inject('md', md)
}
```

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "markdown-it-image-size",
-  "version": "13.2.0-0",
+  "version": "13.2.0-1",
   "description": "Adds width and height to image tags rendered by markdown-it",
   "main": "dist/index.js",
   "files": [

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "markdown-it-image-size",
-  "version": "13.1.1",
+  "version": "13.2.0-0",
   "description": "Adds width and height to image tags rendered by markdown-it",
   "main": "dist/index.js",
   "files": [

--- a/src/index.ts
+++ b/src/index.ts
@@ -14,7 +14,10 @@ export function markdownItImageSize(md: markdownIt): void {
     const otherAttributes = generateAttributes(md, token);
 
     const isExternalImage =
-      imageUrl.startsWith("http://") || imageUrl.startsWith("https://");
+      imageUrl.startsWith("http://") ||
+      imageUrl.startsWith("https://") ||
+      imageUrl.startsWith("//");
+
     const isLocalAbsoluteUrl = imageUrl.startsWith("/");
 
     const { width, height } = isExternalImage
@@ -68,7 +71,9 @@ function getImageDimensionsFromExternalImage(imageUrl: string): {
   width: number;
   height: number;
 } {
-  const response = fetch(imageUrl);
+  const isMissingProtocol = imageUrl.startsWith("//");
+
+  const response = fetch(isMissingProtocol ? `https:${imageUrl}` : imageUrl);
   const buffer = response.buffer();
   const { width, height } = imageSize(buffer);
 

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -48,4 +48,20 @@ describe(markdownItImageSize.name, () => {
 
     expect(actual).toBe(expected);
   });
+
+  it("should render external images with no explicit protocol with attributes for width and height", () => {
+    const markdownRenderer = new MarkdownIt().use(markdownItImageSize);
+
+    const imageUrl =
+      "//images.unsplash.com/photo-1577811037855-935237616bac?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2167&q=80";
+    const markdown = `![](${imageUrl})`;
+
+    const imageWidth = 2167;
+    const imageHeight = 1625;
+
+    const expected = `<p><img src="${imageUrl}" alt="" width="${imageWidth}" height="${imageHeight}"></p>\n`;
+    const actual = markdownRenderer.render(markdown);
+
+    expect(actual).toBe(expected);
+  });
 });

